sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -79,7 +79,7 @@ abstract class DataType extends AbstractDataType {
 
   override private[sql] def defaultConcreteType: DataType = this
 
-  override private[sql] def acceptsType(other: DataType): Boolean = this == other
+  override private[sql] def acceptsType(other: DataType): Boolean = sameType(other)
```

Comment:
cc @cloud-fan 

This is changed to test sameType in order to support ArrayType(StringType)